src/Oceananigans.jl

```diff
@@ -136,11 +136,12 @@ Abstract supertype for fields stored on an architecture `A` and defined on a gri
 abstract type AbstractField{A, G} end

```

Comment:
This is a design question. At the moment we do not need both because all subtypes of `AbstractField` are also subtypes of `AbstractLocatedField`. Thus functionally we could nuke `AbstractLocatedField` and change the definition of `AbstractField` to carry around `X, Y, Z`. 

The only concern is that at some point in the future we may actually want the concept of an "unlocated" field; in that case we would regret such a change. Some "fields" like a constant value, may not naturally have a "location". On the other hand, the field arithmetic introduced in this PR requires a field to have a location; thus even a constant field would have to define a nominal location to participate in abstract operations, even though that location would have no effect on operations.

In summary: probably ok to delete `AbstractLocatedField`. But maybe save for future PR?

Reply:
I guess it's only a one-liner so seems find to keep it the way it is right now.

I agree maybe there will be a use for it in the future (although that's probably not a good reason for keeping around unused code).

Reply:
Such defensive code design is not altogether terrible. It's used in GPUifyLoops for example (the abstract type `GPU` with one subtype, `CUDA`). 

We do actually utilize `AbstractField` to make dispatching on `A` a bit easier, since getting to `A` in `AbstractLocatedField` requires wading through `X, Y, Z`.